liam2/matching.py

```diff
@@ -8,8 +8,10 @@
 from liam2.exprbases import FilteredExpression
 from liam2.context import context_length, context_delete, context_subset, context_keep
 from liam2.utils import loop_wh_progress
-# FIXME: should be optional
-from liam2.cpartition import group_indices_nd
+try:
```

Comment:
the FIXME should stay there as long as we don't provide a pure-python version of group_indices_nd